### PR TITLE
laminar: 0.8 -> 1.0

### PR DESCRIPTION
**Pull request by maralorn**
Bumped version and updated javascript dependencies.
Had to modify the no-network patch as well because of the changed
javascript libs.

Upstream changed the handling CMAKE_INSTALL_PREFIX in
https://github.com/ohwgiles/laminar/commit/1bb545e3f95178aa024f791f6e01510c9d421440
so I decided to patch around that as well.

@andir told me that an enableSystemd option is not necessary. The
systemd files wouldn‘t be installed without user intervention anyways.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by zakame** [comment]
Result of `nixpkgs-review pr 108334` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>laminar</li>
  </ul>
</details>

**Comment by maralorn** [comment]
Mentioning @KAction as maintainer, since that apparently hadn‘t happened automatically.

File: pkgs/development/tools/continuous-integration/laminar/default.nix
Comment (andir) [inline pkgs/development/tools/continuous-integration/laminar/default.nix:57]: I wonder if this is actually useful. Where does laminar look for configuration by default?

This feels like something that should be done in the NixOS modules instead.

Comment (maralorn) [reply pkgs/development/tools/continuous-integration/laminar/default.nix:57]: Well, laminar doesn‘t look into that file at all on its own.
That file contains environment variables supposed to be passed to laminar by systemd.
This option is useful when you want to use the vanilla systemd service that is part of this package.
Since I am planning on upstreaming a module for this, I don‘t think anyone would want to use it, though.

Comment (KAction) [reply pkgs/development/tools/continuous-integration/laminar/default.nix:57]: I am okay with removal of "customConfig" option. Config is not part of laminar itself (unlike, e.g how dwm does it).

**Comment by KAction** [comment]
It does not look to me that you can just drop "enableSystemd" option:

````
(ins)[nix-shell:~/.cache/nixpkgs-review/pr-108334-1]$ find ./results/laminar/|grep service
./results/laminar/lib/systemd/system/laminar.service
````

**Comment by maralorn** [comment]
Why not? Yes, the files are now in there. But do they do any harm?

**Comment by KAction** [comment]
I ask "are they needed", not "do they harm". Probably even better solution would be to put it into separate output.

**Comment by aanderse** [comment]
This `.service` file is a simple text file that can do nothing without full sysadmin privileges. Yes, the `.service` file is needed by anyone who wants to run this as a `systemd` service... A separate output seems unnecessary.

@maralorn @andir are you good to merge this? Aside from `laminard version xx-unversioned started` text spitting out on startup it looks fine to me after doing some simple testing.

**Comment by andir** [comment]
I think we should just remove that upstream systemd unit entirely. There is almost no point in having it in the package as the file doesn't work without spilling files into hardcoded locations (that we have to infer from the upstream unit). We should be fully declarative and just write the units ourselvs.

**Comment by maralorn** [comment]
@aanderse Thanks for picking up on this!
@andir Thanks for the solution.

I have decided to get rid of systemd units and the config file. Because they have no use. I also added myself as a maintainer for good measure.

Is this okay for you @KAction ?

Once this is merged I will approach the quest of upstreaming my laminar module.

**Comment by aanderse** [comment]
Great! Thank you for your work on this.

> Once this is merged I will approach the quest of upstreaming my laminar module.

I look forward to seeing this and the design choices you have made.

File: pkgs/development/tools/continuous-integration/laminar/default.nix
Comment (andir) [inline pkgs/development/tools/continuous-integration/laminar/default.nix:55]: I am find with removing the files if they provide no value. Dow e have proper manpages for the configuration or is the configuration file a good example?

Comment (maralorn) [reply pkgs/development/tools/continuous-integration/laminar/default.nix:55]: That configuration file contains decent documentation for the variables, but it’s completely redundant to the man page `man laminard`.

**Review by aanderse — APPROVED** [review]
LGTM :+1:

Merge?

**Comment by maralorn** [comment]
Let's give @KAction the weekend for a reply. They are the maintainer.

**Comment by KAction** [comment]
Yes, I am perfectly fine with what you came with.

**Comment by aanderse** [comment]
Thanks everyone!

**Pull request merged**
